packages/client/hmi-client/src/components/dataset/tera-column-info.vue

```diff
@@ -0,0 +1,193 @@
+<template>
+	<div class="flex gap-3">
+		<section class="entries">
+			<h6>{{ column.symbol }}</h6>
+			<span class="name">
+				<template v-if="featureConfig.isPreview">{{ column.name }}</template>
+				<tera-input-text
+					v-else
+					placeholder="Add a name"
+					:model-value="column.name ?? ''"
+					@update:model-value="$emit('update-column', { key: 'name', value: $event })"
+				/>
+			</span>
+			<span class="unit">
+				<template v-if="featureConfig.isPreview"><label>Unit</label>{{ column.unit }}</template>
+				<tera-input-text
+					v-else
+					label="Unit"
+					placeholder="Add a unit"
+					:characters-to-reject="[' ']"
+					:model-value="column.unit ?? ''"
+					@update:model-value="$emit('update-column', { key: 'unit', value: $event })"
+				/>
+			</span>
+			<span class="data-type">
+				<label>Data type</label>
+				<template v-if="featureConfig.isPreview">{{ column.dataType }}</template>
+				<Dropdown
+					v-else
+					placeholder="Select a data type"
+					:model-value="column.dataType ?? ''"
+					@update:model-value="$emit('update-column', { key: 'dataType', value: $event })"
+					:options="Object.values(ColumnType)"
+				/>
+			</span>
+			<span v-if="showConcept" class="concept">
+				<label>Concept</label>
+				<template v-if="featureConfig.isPreview">{{ query }}</template>
+				<AutoComplete
+					v-else
+					size="small"
+					placeholder="Search concepts"
+					v-model="query"
+					:suggestions="results"
+					optionLabel="name"
+					@complete="async () => (results = await searchCuriesEntities(query))"
+					@item-select="$emit('update-column', { key: 'concept', value: query })"
+				/>
+			</span>
+			<span class="description">
+				<template v-if="featureConfig.isPreview">{{ column.description }}</template>
+				<tera-input-text
+					v-else
+					placeholder="Add a description"
+					:model-value="column.description ?? ''"
+					@update:model-value="$emit('update-column', { key: 'description', value: $event })"
+				/>
+			</span>
+		</section>
+		<tera-boxplot class="flex-1" v-if="column.stats" :stats="column.stats" />
+	</div>
+</template>
+
+<script setup lang="ts">
+/* Copied the structure of tera-model-parts.vue */
+import { ref, computed, watch } from 'vue';
+import { isEmpty } from 'lodash';
+import TeraInputText from '@/components/widgets/tera-input-text.vue';
+import TeraBoxplot from '@/components/widgets/tera-boxplot.vue';
+import AutoComplete from 'primevue/autocomplete';
+import Dropdown from 'primevue/dropdown';
+import { type DKG, ColumnType, type Grounding } from '@/types/Types';
+import { searchCuriesEntities } from '@/services/concept';
+import type { FeatureConfig } from '@/types/common';
+
+type ColumnInfo = {
+	symbol: string;
+	dataType?: ColumnType;
+	description?: string;
+	grounding?: Grounding;
+	// Metadata
+	unit?: string;
+	name?: string;
```

Comment:
Can we utilize `DatasetColumn` instead of this `ColumnInfo`? I may be misunderstanding but this looks to be a subsection of that?

Also what is the point in symbol and name? Are they not both just `DatasetColumn.name`?
If they are both derived from the same `DatasetColumn.name` how does updating one work?

Reply:
Yeah I tried to do this at first but I had to go for a more UI based approach so the object I mutate have the exact values that I want and are easy to access. Some of the values here are from metadata (see how I construct it in `columnInformation` in `tera-dataset`). I did a similar approach when editing model parts

The `name` is initialized to what the `symbol` is. However it is saved and then read from in `metadata.name` once any edits are applied to it.